doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,201 @@
+vary wildly in nature and conditions.
+Licenses will be approved individually by the Technical Steering Committe (TSC)
+upon request of inclusion of a binary blob.
+Blob vendors may impose a click-through or other EULA-like workflow when users
+fetch and install blobs, but this is not required
+
+Hosting
+=======
+
+Blobs must be hosted on the Internet and managed by third-party infrastructure.
+Two potential examples are Git repositories and web servers managed by
+individual hardware vendors.
+
+The Zephyr Project will not host binary blobs in its Git repositories or
+anywhere else.
+
+Distribution
+============
+
+Distribution happens via a west extension command which users
+can run to fetch binary blobs via the Internet. The extension command
+implementation is hosted in the main zephyr repository. This is required to
+give the project overall control over the mechanism used to fetch the blobs.
+
+The blobs themselves must be specified in the :file:`module.yml` files included
+in separate Zephyr :ref:`module repositories <modules>` maintained by their
+respective vendors.  This means that in order to submit a binary blob to the
+upstream Zephyr distribution, a module repository must exist first or be created
+as part of the submission process.
+
+Each blob which may be fetched must be individually identified in the
+corresponding :file:`module.yml` file. A specification for a blob must contain:
+
+- Purpose: an abstract description of the blob itself
+- A reference to vendor-provided documentation
+- The blob’s type, which must be one of the allowed types
+- A checksum for the blob, which the west extension checks after downloading.
+  This is required for reproducibility and to allow bisecting issues as blobs
+  change using Git and west
+- License text applicable to the blob or a reference to such text
+
+The west extension used to fetch blobs accepts command line arguments that
+allow the user to select the module or modules to fetch blobs from. This is
+required for user control and a uniform user experience regardless of the blob
+vendor.
+
+Allowed types
+=============
+
+The following binary blob types are acceptable in Zephyr:
+
+* Precompiled libraries: Hardware enablement libraries, distributed in
+  precompiled binary form, typically for SoC peripherals. An example could be an
+  enablement library for a wireless peripheral
+* Firmware images: An image containing the executable code for a secondary
+  processor or CPU.  This can be full or partial (typically delta or patch data)
+  and is generally copied into RAM or flash memory by the main CPU. An example
+  could be the firmware for the core running a Bluetooth LE Controller
+* Miscellaneous binary data files. An example could be pre-trained neural
+  network model data
+
+Note that just because a blob has an acceptable type does not imply that it will
+be unconditionally accepted by the project; any blob may be rejected for other
+reasons on a case by case basis.
```

Comment:
This may sound arbitrary. Maybe we can add an typical example of a such blob that would be rejected?

Reply:
done